aws/resource_aws_db_instance.go

```diff
@@ -220,6 +220,49 @@ func resourceAwsDbInstance() *schema.Resource {
 				},
 			},
 
+			"s3_import": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				ConflictsWith: []string{
+					"snapshot_identifier",
+					"replicate_source_db",
+				},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bucket_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							Optional: false,
+							ForceNew: true,
+						},
+						"bucket_prefix": {
+							Type:     schema.TypeString,
+							Required: false,
+							Optional: true,
+							Default:  "",
+						},
+						"ingestion_role": {
+							Type:     schema.TypeString,
+							Required: true,
+							Optional: false,
+						},
+						"source_engine": {
+							Type:     schema.TypeString,
+							Required: false,
+							Optional: true,
+							Default:  "mysql",
```

Comment:
If/when this is supported for other database engines, this will become awkward with having engine and engine version defaults. I think we should set both these two `Required: true` (like the API), remove the `Default:`s and document the valid values for each attribute for now.